### PR TITLE
feat(api): GiveAPIServer returns the mux with API, denyRPC as its own flag

### PR DESCRIPTION
**Pull request by dustmop**

### CODE REVIEW

**Comment by dustmop** [comment]
fixes https://github.com/qri-io/qri/issues/1759

File: lib/routes.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+func (inst *Instance) GiveAPIServer(middleware func(handler http.HandlerFunc) http.HandlerFunc, ignoreMethods []string) *mux.Router {
+	m := mux.NewRouter()
+	for methodName, call := range inst.regMethods.reg {
+		if arrayContainsString(ignoreMethods, methodName) {
+			continue
+		}
+		handler := middleware(NewHTTPRequestHandler(inst, methodName))
+		// All endpoints use POST verb
+		httpVerb := http.MethodPost
+		m.Handle(string(call.Endpoint), handler).Methods(httpVerb)
```
Comment (Arqu) [inline lib/routes.go:19]: While we're here, can we also cover for `OPTIONS` requests as CORS chokes on it.

**Review by Arqu — APPROVED** [review]
This is awesome. All routes are correct, and lovely to see the dispatch pattern creep up all the way into `api`.
`denyHTTP` solves a piece of docs more elegantly now too. 

Except for the `OPTIONS` comment, LGTM.

File: api/api.go [hunk 2]
```diff
@@ -206,10 +206,7 @@ func handleRefRoute(m *mux.Router, p refRouteParams, f http.HandlerFunc) {
 func NewServerRoutes(s Server) *mux.Router {
 	cfg := s.GetConfig()
 
-	m := s.Mux
-	if m == nil {
-		m = mux.NewRouter()
-	}
+	m := s.Instance.GiveAPIServer(s.Middleware, []string{"dataset.get"})
 	m.Use(corsMiddleware(cfg.API.AllowedOrigins))
```
Comment (Arqu) [inline api/api.go:210]: This line will take the lib supplied mux and allow CORS given OPTIONS requests pass to it.

**Pull request merged**